prime-router/docs/authz/authz-design.md

```diff
@@ -0,0 +1,279 @@
+settings within the RS Postgres DB.
+
+Example: User within the `elims` organization wants to see the `md-phd` daily data page to check that reports have 
+been properly routed.
+
+Organizational settings:
+```yml
+md-phd:
+  allow-daily-data-access: [elims] # references another organization name
+```
+
+User token:
+```json
+{
+  "scp": [
+    "openid",
+    "email",
+    "org:write"
+  ],
+  "org": [
+    "elims"
+  ]
+}
+```
+When a user tries to access an organization's daily data page, we would need to compare that user's `org` claim against 
+the values configured for that organization. In the example above, the user would have access to the `md-phd` daily data 
+page because it was configured to allow it for `elims` users.
+
+This will allow organizations to have control over whether their data is shown within the daily data page as an opt-in 
+feature rather than exposing their data without their knowledge.
+
+## Setting up a new user
+
+In this system, setting up a new user would be quite easy. Create the user in Okta and then add it to the required 
+groups. If this particular user must also manually submit reports you will need to directly modify their profile to add 
+the specific senders. For users this can be done in the Okta UI.
+
+### Difficulty with sender setup
+Senders are set up as application users within Okta to allow the machine-to-machine client credentials OAuth 2.0 flow. The 
+difficulty arises because while Okta allows you to add application users to groups, it does not allow you to easily add 
+those groups to the generated access token JWT. The workaround there is adding the necessary information to the Okta 
+application user's profile. This can only be done via API at the moment though.
+
+My suggestion is having a sender setup API and CLI that when given the appropriate values, will set them in the profile.
+
+Here is some psudeocode on what that might look like using the Okta admin SDK. Given this would happen in the `auth` 
+project we could easily authorize that the user kicking off this job only be a superadmin.
+```kotlin
+fun setApplicationProfile(
+    clientId: String,
+    submit: List<String>
+) {
+    val application = applicationApi.getApplication(clientId, null)
+    val groups = getApplicationGroups(clientId)
+    application.putprofileItem("groups", groups)
+    application.putprofileItem("appSubmit", submit)
+    applicationApi.replaceApplication(arnejAppId, application)
+}
+```
+
+We could go even further and write various APIs (and potentially CLI calls) that will set up an application user from 
+scratch. This would dramatically simplify the process of setting up a sender since we would not need to handle going
+back and forth between API and Okta Admin UI and could provide the necessary values needed for a new sender.
+We could also reuse our existing authentication and authorization within the project to ensure only specific people are able to 
+hit these endpoints.
```

Comment:
Maybe just call out that the next/final step could be to create a GUI in ReportStream where engagement can fill out a form and the app user is correctly created/updated.